pkg/lib/long-running-process.js

```diff
@@ -68,7 +69,13 @@ export default class LongRunningProcess {
             this.setRunning(true);
         } else if (activeState === 'failed') {
             this.startTimestamp = null; // TODO: can we derive this from InvocationID?
-            this.setRunning("failed");
+            if (this.terminated) {
+                /* terminating causes failure; reset that and do not announce it as failed */
+                this.systemdClient.call(O_SD_OBJ, I_SD_MGR, "ResetFailedUnit", [this.serviceName], { type: "s" })
+            } else {
+                /* If you don't care about "failed" state (not from termination), call systemd-run with --collect */
```

Comment:
This comment sounds odd here.  How would one call systemd-run with --collect without changing the source code?

Reply:
Right now you can't -- a project that wants to use the library needs to copy it anyway, and could adjust it. I'm not sure I want to parameterize that for now, as it seems cleaner to handle failures. As this is a step-by-step improvement, I could just remove the comment for now, WDYT?

Reply:
Removing the comment sounds good.